src/praisonai-agents/praisonaiagents/gateway/protocols.py

```
@@ -302,6 +302,357 @@ def to_dict(self) -> Dict[str, Any]:
+            if not value.is_integer():
+                raise ValueError
+            return int(value)
+        return int(value)
+    except (TypeError, ValueError):
+        raise FrameDecodeError(
+            HelloError(
+                code=ConnectErrorCode.CONFIGURATION_ERROR,
+                message=f"Field '{field_name}' must be an integer, got {value!r}",
+                next_step=ConnectRecoveryStep.DO_NOT_RETRY,
+            )
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)